### ISSUE TITLE
Question about paper's equation

### ISSUE THREAD

**Issue by junseokoh1** [taldatech/soft-intro-vae-pytorch#16]
Thanks for your nice work and let us to use code.
I have some question about paper. As Soft-IntroVAE is modified IntroVAE i read introVAE. At section 3, which is background, i think equation (2) is different from IntroVAE.

![image](https://user-images.githubusercontent.com/28286337/163574319-3cf01a54-66a3-44eb-af37-e38fee3a1d26.png)

This equation is from IntroVAE paper where E(x) = D_{KL}(q_φ(z|x)||p(z))
![image](https://user-images.githubusercontent.com/28286337/163574105-cd9ae157-893a-4957-bce4-baeae98c7af5.png)
And this equation is from Soft-IntroVAE paper. I think L_{E_φ}(x, z) is same with introVAE, but L_{D_θ} is different from introVAE equation with additional E(x) term.
If you don't mind can you explain about this? sorry for the bad english

**Comment by taldatech** [comment]
Hi, thank you for the interest in our paper.
The equations are the same, as we explain in the paper:
For the decoder/generator, IntroVAE optimizes the reconstruction error, L_{AE}(x) = L_{rec}(x) and the KL divergence as calculated from the output of the encoder, KL(E(D(z))) = E(G(z)) (in IntroVAE). Recall that the ELBO for real data is comprised of the sum: ELBO(X) = L_rec(x) + KL. However, the KL divergence (for real data) only updates the encoder, so for the decoder loss (L_D) we can just add it, but it has no effect (just a constant), thus, we get that the decoder also optimizes the ELBO of real data (our paper formulation), but only the reconstruction error actually updates its parameters and thus omitted in IntroVAE.

**Comment by junseokoh1** [comment]
Thanks to you, I understood.
Thank you for your kind explanation.

**Comment by junseokoh1** [comment]
I'm sorry for keep asking about paper's equations.
I'm little confuse about what P_{d*}(z|x) mean which is 4.1 Analysis
![image](https://user-images.githubusercontent.com/28286337/163707334-8bf6757a-1a96-4df1-8c99-37bd2608c5fe.png)
Is this mean latent vector from encoder where x is generated image from optimal decoder?

**Comment by taldatech** [comment]
This is the optimal posterior (=optimal encoder) for the optimal decoder. In the VAE framework, we use an approximated posterior, q(z|x), and we ideally want to minimize its distance to the optimal posterior p(z|x) (which we don't know).

**Comment by junseokoh1** [comment]
So p_{d*}(z|x) is same with q(z|x) when x is generated image from optimal decoder? And by the nash equilibrium this is also optimal posterior.

**Comment by taldatech** [comment]
Yes,  if I understood correctly.

**Comment by junseokoh1** [comment]
I'm sorry for keep bothering you... I want to fully understand your paper but it is hard for me...
I also keep looking at the appendix too. It's a little bit hard to follow the flow of proof.
I can't understand why d* is defined like equation (6). 
![image](https://user-images.githubusercontent.com/28286337/163830173-1c86c572-1acb-4d0c-8ee3-2d05a16179bf.png)
Yes by defining d* as equation (6) then (q*, d*) is a Nash equilibrium of equation (5). 
I wonder if the reason for that definition is derived from other equations.

**Comment by taldatech** [comment]
For the theoretical analysis, please see section 9.6 in the appendix.
Regarding your question. see the derivation in Eq. 17 in the appendix, where we show why this is true.

**Comment by junseokoh1** [comment]
Thank you for your kindness.
